File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/factory/transport/Bike.java
// Bike.java

package com.bigboxcode.designpattern.factory.transport;

public class Bike implements Transport {
    @Override
    public void start() {
        System.out.println("Bike started");
    }

    @Override
    public void stop() {
        System.out.println("Bike Stopped");
    }

    @Override
    public void repair() {
        System.out.println("Bike Repair");
    }
}

File: design-patterns/ts/factory/transport-factory-method/bike.ts
// bike.ts

import Transport from "../transport/transport";

class Bike implements Transport {
    start(): void {
        console.log("Bike started");
    }

    stop(): void {
        console.log("Bike Stopped");
    }

    repair(): void {
        console.log("Bike Repair");
    }
}

export default Bike;